src/switch_rtp.c

```diff
@@ -2049,6 +2050,78 @@ static int using_ice(switch_rtp_t *rtp_session)
 	return 0;
 }
 
+static void switch_send_rtcp_event(switch_rtp_t *rtp_session ,struct switch_rtcp_sender_report *sr,struct switch_rtcp_report_block *rtcp_report_block) {
+	if (sr && rtcp_report_block) {
+		switch_event_t *event;
+
+		if (switch_event_create(&event, SWITCH_EVENT_SEND_RTCP_MESSAGE) == SWITCH_STATUS_SUCCESS) {
+				char value[30];
+				char header[50];
+				uint32_t tmpLost;
+				char *uuid = switch_core_session_get_uuid(rtp_session->session);
+				if (uuid) {
+						switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Unique-ID", switch_core_session_get_uuid(rtp_session->session));
+				}
+
+				snprintf(value, sizeof(value), "%.8x", rtp_session->stats.rtcp.ssrc);
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "SSRC", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.ntp_msw));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "NTP-Most-Significant-Word", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.ntp_lsw));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "NTP-Least-Significant-Word", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.ts));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "RTP-Timestamp", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.pc));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Sender-Packet-Count", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.oc));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Octect-Packet-Count", value);
+
+				snprintf(value, sizeof(value), "%u", ntohl(sr->sender_info.ts));
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Last-RTP-Timestamp", value);
+
+				snprintf(value, sizeof(value), "%" SWITCH_TIME_T_FMT, switch_time_now());
+				switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Capture-Time", value);
+
+				// Add sources info
+				snprintf(header, sizeof(header), "Source-SSRC");
+				snprintf(value, sizeof(value), "%.8x", rtcp_report_block->ssrc);
```

Comment:
This still isn't correct, because rtcp_report_block->ssrc is written with network byte ordering, and we want host for this sprintf method.  From rtcp_generate_report_block():
`rtcp_report_block->ssrc = htonl(rtp_session->stats.rtcp.peer_ssrc);`
Sorry I didn't catch that sooner.  So the line this comment refers to should be:
`snprintf(value, sizeof(value), "%.8x", rtp_session->stats.rtcp.peer_ssrc);`

Reply:
@rjm2718, thank you so much for finding this.  Branch is updated to reflect this, and rebased from master.